### DISCUSSION TITLE
setting up simple app to show elements from a table

### DISCUSSION THREAD

**Discussion by markb9** [supabase/supabase#5925] [Questions]
hello all, i'm trying to use supabase to create an app with expo/react native, i'm a beginner and i'm trying to understand how to get data from tables and columns, when compiling i get an error saying that it failed to render here's a screenshot of the code: 
![image](https://user-images.githubusercontent.com/100837913/158349589-1b839538-780a-421a-8c11-5f78e43b624d.png)

**Comment by Hallidayo** [comment]
Hi,

Have you tried using the useEffect hook to load the data on when the view renders?

I would also move the function `read` inside your `App` function as well and also creating a separate file for your Supabase client.

It may be worth looking at [this](https://supabase.com/docs/guides/with-expo#initialize-a-react-native-app) guide on the Supabase website. On the Accounts page section it shows you how to SELECT from your Supabase database using React Native/Expo.
